addons/mail/controllers/thread.py

```diff
@@ -169,3 +169,9 @@ def mail_message_update_content(self, message_id, body, attachment_ids, attachme
 
     def _is_message_editable(self, message, **kwargs):
         return message.sudo().is_current_user_or_guest_author or request.env.user._is_admin()
+
+    @http.route("/mail/thread/unsubscribe", methods=["POST"], type="jsonrpc", auth="user")
+    def mail_thread_unsubscribe(self, res_model, res_id, partner_ids):
+        thread = self.env[res_model].browse(res_id)
+        thread.message_unsubscribe(partner_ids)
+        return Store(thread, [], as_thread=True, request_list=["followers"]).get_result()
```

Comment:
There are still many extra RPC done after removing a follower.
data, messages, web_read and data again...
At least the first data (with followers and suggestedRecipients) should be replaced by this line.
I don't think messages is necessary in the flow of removing a follower.
`web_read` I would say it depends if any field in the view depends on follower, which is a bit complicated to know.
And data again (with all request params this time) I'm not sure what's the point.

Reply:
It looks like what we lack is a specific version of `onFollowerChanged` but only for when removing one, which requires less processing.

Note that ideally `this.load(thread, ["followers", "suggestedRecipients"]);` should be removed in all flows, and returned by the RPC adding/updating the followers instead, similar to what is done here for removal.